source/common/network/connection_impl.cc

```diff
@@ -369,7 +378,7 @@ void ConnectionImpl::readDisable(bool disable) {
       // buffer's high watermark has triggered.
       ASSERT(read_buffer_.length() > 0 || read_disable_count_ == 0);
       dispatch_buffered_data_ = true;
-      setReadBufferReady();
+      ioHandle().activateFileEvents(Event::FileReadyType::Read);
```

Comment:
Think it's worth a comment on why this case is different?

Reply:
Rearranged and expanded comments.